paddle/fluid/framework/ir/mkldnn/matmul_transpose_reshape_mkldnn_fuse_pass.cc

```diff
@@ -84,6 +84,12 @@ void MatmulTransposeReshapeMKLDNNPass::Fuse(
     }
 
     OpDesc *matmul_desc = matmul_op->Op();
+    matmul_desc->SetType("fused_matmul");
```

Comment:
There have some `broadcast`  computational logic difference between `matmul_v1` and `matmul_v2`， the code as follows

```python
import paddle 
data1 = paddle.rand((3, 1, 10, 10))
data2 = paddle.rand((1, 2, 10, 10))


# The paddle.matmul (matmul_v2)
paddle.matmul(data1, data2).shape # [3, 2, 10, 10]


# The fluid.layers.matmul (matmul_v1)
paddle.fluid.layers.matmul(data1, data2).shape # 直接报错
# ValueError: (InvalidArgument) The batch size of the two matrices should be equal, or at least one is zero.
# But received X's shape: [3, 10, 10], Y's shape: [2, 10, 10]
```
Is there some problems when replacing directly ?

Reply:
Hi, I don't think that it could be problem. For some time oneDNN version of both `matmul` and `matmul_v2` are already using same code https://github.com/PaddlePaddle/Paddle/pull/44640, https://github.com/PaddlePaddle/Paddle/pull/48162. Declared attributes of  these two are mostly the same. Only difference is `alpha` not supported in `matmul_v2`, and `transpose` is renamed to `trans`).

Goal of this PR is just to extract fusion logic from base op and base kernel. Fused op is superset, which has all extra attributes declared in OPMaker and fused kernel has implemented support for handling  fusion logic. 

I've adjusted all fuse pass unit test to work with `fused_matmul`. Also this PR has been checked in our internal validation and it didn't report any accuracy drop.